MdeModulePkg/Core/Dxe/Mem/Page.c

```diff
@@ -141,6 +151,40 @@ RemoveMemoryMapEntry (
   }
 }
 
+EFI_MEMORY_TYPE
+GetBucketMemoryType (
+  IN EFI_PHYSICAL_ADDRESS  PhysicalStart,
+  IN EFI_PHYSICAL_ADDRESS  PhysicalEnd
+  )
+{
+  EFI_MEMORY_TYPE  BucketType;
+
+  // Find the bucket type for the incoming memory region.
+  for (BucketType = 0; BucketType < EfiMaxMemoryType; BucketType++) {
+    //
+    // If the number of pages for this memory type is not zero, the input region
+    // better be within the same bucket.

```

Comment:
Update the comment for why the `Special` field is used?

Reply:
Updated.